### DISCUSSION TITLE
How to use k8s to dynamically expand capacity?

### DISCUSSION THREAD

**Discussion by gaecoli** [getredash/redash#7175] [Q&A]
Is there a way to use k8s expand capacity? While a large of users to use redash, the workers will stuck at queue statu.

**Comment by rcheatham-q** [comment]
This is something I have toyed with and I believe it's possible using [KEDA](https://keda.sh/). You should be able to configure KEDA to read from your Celery broker (Redis or RabbitMQ) and scale your workers appropriately. You may need to create one deployment per queue to get full scalability, though. That will depend on the relative load of ad-hoc and scheduled queries.

**Reply by rcheatham-q** [reply]
Two seconds after writing this I discovered [Scaling Modifiers](https://keda.sh/docs/2.15/concepts/scaling-deployments/#scaling-modifiers) which will allow a single KEDA ScaledObject to set the number of workers based on the total length of all queues.

**Reply by justinclift** [reply]
Bear in mind that modern Redash uses RQ, not Celery.  No idea if that really changes things though. :wink:

**Reply by rcheatham-q** [reply]
It doesn't, RQ uses Redis lists which is a [builtin scaler](https://keda.sh/docs/2.15/scalers/redis-lists/) to KEDA
